### PR TITLE
fix: going back in the browser doesn't update the settings of the experiment list view [DET-5625]

### PR DESCRIPTION
**Pull request by apizzini**
## Description

I spend some time trying to support "reloading" the previous state of the table when going back in the browser history.

That's not easy due to having state partially "synced" between ant-table and the page component (the cleanest solution would have been to update the state by parsing the url when it changes, by avoiding to change the state directly from the component and changing it only by parsing the url).

But this simple fix is probably the best one (avoiding to push a new state in the browser history any time the URL changes but you're still in the experiment list page).

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/ExperimentList.tsx [hunk 1]
```diff
@@ -139,7 +139,7 @@ const ExperimentList: React.FC = () => {
       searchParams.append('user', URL_ALL);
     }
 
-    window.history.pushState(
+    window.history.replaceState(
```
Comment (hkang1) [inline webui/react/src/pages/ExperimentList.tsx:142]: @apizzini the crux of this problem might be more related to the fact that table page offset is not stored in the URL. We intentionally made that choice of not including table page offset in the URL, So when the user clicks a bunch of different table pages, the URL is all the same, so when the user clicks the back button it just defaults to table page offset of 0 (first table page).

What do you think of the solution to add table page offset to the URL query params as well? That way with the pushing of routes, it'll store the page offset info properly and the back button should naturally behave as expected.

**Review by hkang1 — COMMENTED** [review]
@apizzini the crux of this problem might be more related to the fact that table page offset is not stored in the URL. We intentionally made that choice of not including table page offset in the URL, So when the user clicks a bunch of different table pages, the URL is all the same, so when the user clicks the back button it just defaults to table page offset of 0 (first table page).

What do you think of the solution to add table page offset to the URL query params as well? That way with the pushing of routes, it'll store the page offset info properly and the back button should naturally behave as expected.

**Comment by apizzini** [comment]
Table offset and limit are currently in the URL.

http://localhost:3000/det/experiments?...&limit=10&offset=30&...

**Comment by hkang1** [comment]
> Table offset and limit are currently in the URL.
> 
> http://localhost:3000/det/experiments?...&limit=10&offset=30&...

🤦 ah that's right.

Let me take this one as I am working on a [PR](https://github.com/determined-ai/determined/pull/2740) that is related to this area and it will address this issue.

**Comment by hkang1** [comment]
If [PR 2740](https://github.com/determined-ai/determined/pull/2740) looks ok, we can close this PR.

**Pull request closed without merging**